src/main/java/de/numcodex/feasibility_gui_backend/query/QueryHandlerRestController.java

```diff
@@ -26,7 +26,7 @@
 */
 @RequestMapping("api/v1/query-handler")
 @RestController
-@CrossOrigin
+@CrossOrigin(origins = "*", exposedHeaders = "Location")
```

Comment:
This enables CORS for the whole controller - is this desired? If not then please move this to the appropriate endpoint method.

Furthermore `origins = "*"` opens this up to the whole world. If you know where the request is coming from, then please insert the origin here. Always be as restrictive as possible since this is a security feature!
If the origin is not known at this point, then feel free to check whether an upstream proxy can add this header for you (also bypasses any issues of potentially pinned values within the code base).